<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1173</width>
    <height>907</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Компьютерная графика. Лабораторная №8. Отсекатель</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="4" column="1">
     <widget class="QGroupBox" name="groupBox_4">
      <property name="maximumSize">
       <size>
        <width>332</width>
        <height>80</height>
       </size>
      </property>
      <property name="title">
       <string>Канвас</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="0" column="0">
        <widget class="QPushButton" name="pushButton_undo">
         <property name="text">
          <string>Отменить</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QPushButton" name="pushButton_clear">
         <property name="text">
          <string>Очистить</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="1" column="1">
     <widget class="QGroupBox" name="groupBox_2">
      <property name="maximumSize">
       <size>
        <width>332</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string>Отрезки</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_8">
       <item row="0" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Новый отрезок добавляется ЛКМ по холсту</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QGroupBox" name="groupBox_3">
         <property name="title">
          <string>Добавить отрезок с клавиатуры:</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="1" column="1">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_line_x_b">
            <property name="minimum">
             <double>-5000.000000000000000</double>
            </property>
            <property name="maximum">
             <double>5000.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="3" column="2">
           <widget class="QLabel" name="label_7">
            <property name="maximumSize">
             <size>
              <width>12</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>Y:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="label_19">
            <property name="maximumSize">
             <size>
              <width>12</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>X:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QLabel" name="label_6">
            <property name="maximumSize">
             <size>
              <width>12</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>Y:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="0" colspan="2">
           <widget class="QLabel" name="label_5">
            <property name="text">
             <string>Точка начала:</string>
            </property>
           </widget>
          </item>
          <item row="3" column="1">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_line_x_e">
            <property name="minimum">
             <double>-5000.000000000000000</double>
            </property>
            <property name="maximum">
             <double>5000.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="2" column="0" colspan="2">
           <widget class="QLabel" name="label_17">
            <property name="text">
             <string>Точка конца:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="3">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_line_y_b">
            <property name="minimum">
             <double>-5000.000000000000000</double>
            </property>
            <property name="maximum">
             <double>5000.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="3" column="3">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_line_y_e">
            <property name="minimum">
             <double>-5000.000000000000000</double>
            </property>
            <property name="maximum">
             <double>5000.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="3" column="0">
           <widget class="QLabel" name="label_18">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>0</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>12</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>X:</string>
            </property>
           </widget>
          </item>
          <item row="1" column="5" rowspan="3">
           <widget class="QPushButton" name="pushButton_line_add">
            <property name="minimumSize">
             <size>
              <width>0</width>
              <height>80</height>
             </size>
            </property>
            <property name="maximumSize">
             <size>
              <width>30</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>+</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QTableWidget" name="tableWidget_lines">
         <property name="maximumSize">
          <size>
           <width>304</width>
           <height>16777215</height>
          </size>
         </property>
         <property name="font">
          <font>
           <pointsize>7</pointsize>
           <bold>false</bold>
          </font>
         </property>
         <property name="horizontalScrollBarPolicy">
          <enum>Qt::ScrollBarAlwaysOff</enum>
         </property>
         <property name="editTriggers">
          <set>QAbstractItemView::NoEditTriggers</set>
         </property>
         <property name="selectionMode">
          <enum>QAbstractItemView::NoSelection</enum>
         </property>
         <column>
          <property name="text">
           <string>Нач. точка</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Кон. точка</string>
          </property>
         </column>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="2" column="1">
     <widget class="QGroupBox" name="groupBox_5">
      <property name="maximumSize">
       <size>
        <width>332</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="title">
       <string>Отсекатель</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_6">
       <item row="3" column="0">
        <widget class="QPushButton" name="pushButton_cutter_connect">
         <property name="text">
          <string>Замкнуть</string>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QPushButton" name="pushButton_cut">
         <property name="text">
          <string>Отсечь</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0" colspan="2">
        <widget class="QLabel" name="label_14">
         <property name="text">
          <string>Отсекатель меняется нажатием ПКМ на холсте</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0" colspan="2">
        <widget class="QGroupBox" name="groupBox_6">
         <property name="title">
          <string>Добавить вершину отсекателя</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_2">
          <item row="0" column="0">
           <widget class="QLabel" name="label_21">
            <property name="maximumSize">
             <size>
              <width>12</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>X:</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_box_x_b">
            <property name="minimum">
             <double>-5000.000000000000000</double>
            </property>
            <property name="maximum">
             <double>5000.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QDoubleSpinBox" name="doubleSpinBox_box_y_b">
            <property name="minimum">
             <double>-5000.000000000000000</double>
            </property>
            <property name="maximum">
             <double>5000.000000000000000</double>
            </property>
           </widget>
          </item>
          <item row="0" column="4" rowspan="2">
           <widget class="QPushButton" name="pushButton_box_add">
            <property name="maximumSize">
             <size>
              <width>30</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>+</string>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QLabel" name="label_16">
            <property name="maximumSize">
             <size>
              <width>12</width>
              <height>16777215</height>
             </size>
            </property>
            <property name="text">
             <string>Y:</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="2" column="0" colspan="2">
        <widget class="QTableWidget" name="tableWidget_cutter">
         <column>
          <property name="text">
           <string>X</string>
          </property>
         </column>
         <column>
          <property name="text">
           <string>Y</string>
          </property>
         </column>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="1">
     <widget class="QGroupBox" name="groupBox">
      <property name="maximumSize">
       <size>
        <width>332</width>
        <height>133</height>
       </size>
      </property>
      <property name="title">
       <string>Настройки цвета</string>
      </property>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="3" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Цвет отсечения:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Цвет отрезков:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Цвет фона:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QPushButton" name="pushButton_color_bg">
         <property name="styleSheet">
          <string notr="true">background-color: rgb(255, 255, 255);</string>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QPushButton" name="pushButton_color_line">
         <property name="styleSheet">
          <string notr="true">background-color: rgb(0, 0, 0);</string>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QPushButton" name="pushButton_color_cutted">
         <property name="styleSheet">
          <string notr="true">background-color: rgb(0, 170, 0);</string>
         </property>
         <property name="text">
          <string/>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
    </item>
    <item row="0" column="2" rowspan="6">
     <widget class="QGraphicsView" name="graphicsView"/>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1173</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menu">
    <property name="title">
     <string>Инфо</string>
    </property>
    <addaction name="about_programm"/>
    <addaction name="about_author"/>
   </widget>
   <addaction name="menu"/>
  </widget>
  <action name="about_programm">
   <property name="text">
    <string>О программе</string>
   </property>
  </action>
  <action name="about_author">
   <property name="text">
    <string>Об авторе</string>
   </property>
  </action>
 </widget>
 <resources/>
 <connections/>
</ui>
